packages/ember-data/tests/integration/filter-test.js

```diff
@@ -16,8 +16,34 @@ var shouldNotContain = function(array, item) {
 
 module("integration/filter - DS.Model updating", {
   setup: function() {
-    array = [{ id: 1, name: "Scumbag Dale", bestFriend: 2 }, { id: 2, name: "Scumbag Katz" }, { id: 3, name: "Scumbag Bryn" }];
-    Person = DS.Model.extend({ name: DS.attr('string'), bestFriend: DS.belongsTo('person', { inverse: null, async: false }) });
+    array = [{
+      "id": "1",
+      "type": "person",
+      "attributes": {
+        "name": "Scumbag Dale"
+      },
+      relationships: {
+        bestFriend: {
+          data: {
+            id: 2,
+            type: "person"
+          }
+        }
+      }
+    }, {
+      "id": "2",
+      "type": "person",
+      "attributes": {
+        "name": "Scumbag Katz"
+      }
+    }, {
+      "id": 3,
```

Comment:
Nitpicking here but as we want to conform to JSON-API as closely as possible IDs should be strings (http://jsonapi.org/format/#document-resource-object-identification).

There are a couple of places below too where IDs are numbers.